### PR TITLE
FF114 Popover API behind preference

### PR DESCRIPTION
**Pull request by hamishwillee**
FF114 supports popover Javascript API in https://bugzilla.mozilla.org/show_bug.cgi?id=1823757 behind the pref `dom.element.popover.enabled`

This includes:
- [`HTMLButtonElement.popoverTargetAction`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLButtonElement/popoverTargetAction)
- [`HTMLButtonElement.popoverTargetElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLButtonElement/popoverTargetElement)
- [`HTMLInputElement.popoverTargetElement`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/popoverTargetElement),
- [`HTMLInputElement.popoverTargetAction`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/popoverTargetAction) 
- [`HTMLElement.popover`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/popover) 
- [`HTMLElement.hidePopover()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/hidePopover) 
- [`HTMLElement.showPopover()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/showPopover) 
- [`HTMLElement.togglePopover()`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/togglePopover) 
- `HTMLElement` [`beforetoggle` event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/beforetoggle_event) 
- `HTMLElement` [`toggle_event` event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/toggle_event)
- [`ToggleEvent`](https://developer.mozilla.org/en-US/docs/Web/API/ToggleEvent) 
- CSS selector [`::backdrop`](https://developer.mozilla.org/en-US/docs/Web/CSS/::backdrop)
- CSS selector [`:popover-open`](https://developer.mozilla.org/en-US/docs/Web/CSS/:popover-open)
- HTML global attribute: [`popovertarget`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#popovertarget)
- HTML global attribute: [`popovertargetaction`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#popovertargetaction)

Related docs work can be tracked in https://github.com/mdn/content/issues/26694

### CODE REVIEW

**Review by Elchi3 — CHANGES_REQUESTED** [review]
What about updating these?

- api.HTMLElement.toggle_event
- css.selectors.backdrop.popover
- css.selectors.popover-open
- html.global_attributes.popovertarget
- html.global_attributes.popovertargetaction

**Review by Elchi3 — COMMENTED** [review]
Meant that as a comment, if the rest of the API is not supported then this PR is fine.

**Comment by hamishwillee** [comment]
Thanks @Elchi3 All added.

TLDR;
- I missed  `api.HTMLElement.toggle_event` because I relied on it being [decorated with  `dom.element.popover.enabled`](https://searchfox.org/mozilla-central/source/dom/webidl/EventHandler.webidl#105). 
- Normally we separate out APIs, HTML, CSS work in an FF release project. But makes sense here as there seem to be no other docs projects for tracking those parts of the API. I have set version as 114 as none of this is useful without that preference enabled and the ability to launch the popup.

**Review by Elchi3 — APPROVED** [review]
Thanks @hamishwillee! 👍 I think this is a prime example for how the https://github.com/web-platform-dx/feature-set project will help us as well to document all features of a thing.

**Comment by hamishwillee** [comment]
Thank you. That's an ambitious project, with some BCD overlap. Will be interesting to see how it pans out with the partial implementation of things that we often get across browsers. But yes, that will be useful.
